draft-ietf-tls-external-psk-importer.md

```diff
@@ -138,14 +178,70 @@ EPSKs may be imported for early data use if they are bound to protocol settings
 otherwise be required for early data with normal (ticket-based PSK) resumption. Minimally, that means ALPN,
 QUIC transport settings, etc., must be provisioned alongside these EPSKs.
 
+##Client ID {#client-id-description}
+The `client_id` is an optional field that is required to be unique for each actor that knows the EPSK.
+This is to prevent Selfie-style reflections.
+If two actors use the same `client_id` this defence may not be effective. 
+This is only necessary in scenarios where more than two actors use the same key, including the case where a single agent will complete PSK handshakes as both the client and the server using the same key.
+
+The Selfie attack {{Selfie}} abuses an underlying assumption of TLS, that only one client and one server know a PSK.
+A pair of agents that both act as both client *and* server, and use the same PSK in both roles violate this assumption.
+In this case, there are two clients and two servers that know the PSK, since each endpoint acts as both client and server.
+
+Because both clients have identical configurations, an agent acting as a server can't distinguish between its peer acting as a client, and itself acting as a client.
+The agents are thus both vulnerable to reflection attacks where the attacker forces the agent to act as both client and server in a single connection.
+Adding a role-based label is ineffective, as both the agent and its peer will add the `client` label when acting as a client, leaving them indistinguishable.
+
+Recall both clients have identical configurations.
+Therefore even though the session specific parameters change for each session, the server cannot distinguish between a `ClientHello` reflected back on it by the attacker and a `ClientHello` from its peer.
+
+Broadly speaking, there are two types of parameters in the TLS handshake, those which are constant between sessions, and those which change with every session.
+Parameters that are constant between sessions can be used to 'fingerprint' peers, however, if two agents have the same configuration, they will have the same 'fingerprint'.
+Parameters that change with each session cannot be used to distinguish between two peers, because the values are always unrelated.
+
+By hashing a distinguishing `client_id` value into the PSK binder, a server that received a reflected `ClientHello` would be unable to verify the binder, and would reject the connection.
+This specifically breaks the symmetry of the configurations such that agents are distinguishable.
+
+For example, distinguishing strings for agents in peer-to-peer IoT device deployments could be a MAC address.
+Similarly, distinguishing strings for agents in mesh VM deployments could be namespaces.
+The decision to use this field and what it should contain MUST be agreed OOB.
+Note that, critically, the TLS protocol never sends this value on the wire, as this would identify the client even to passive adverseries.
```

Comment:
```suggestion
Note that, critically, the TLS protocol never sends `client_id` on the wire, as this would identify the client even to passive adversaries.
```

Reply:
Technically the `client_id` is (symbolically) sent on the wire, as it is included in the PSK binder, however it is blinded by the `epsk`. If it were unblinded but hashed then it could still be used for fingerprinting.